DIAGRAMS.md

```diff
@@ -0,0 +1,304 @@
+# Diagrams
+
+# Project Structure
+Kyoo is a monorepo that consists of several projects each in their own directory.  Diagram below shows an outline of kyoo, projects, and artifacts.
+
+```mermaid
+block-beta
+  columns 1
+  block:proj1:1
+    proj_name["Kyoo"]:1
+  end
+  block:proj2:1
+    dir_1["autosync/"]
+    dir_2["back/"]
+    dir_3["front/"]
+    dir_4["transcoder/"]
+    dir_5["scanner/"]
+  end
+  block:proj3:1
+    %% columns auto (default)
+    block:autosync_b:1
+      autosync_i1("kyoo_autosync")
+    end
+    block:back_b:1
+      columns 1
+      back_i1("kyoo_back")
+      back_i2("kyoo_migrations")
+    end
+    block:front_b:1
+      front_i1("kyoo_front")
+    end
+    block:transcoder_b:1
+      transcoder_i1("kyoo_transcoder")
+    end
+    block:scanner_b:1
+      columns 1
+      scanner_i1("kyoo_scanner")
+      scanner_i2("kyoo_scanner*")
+    end
+  end
+
+  style proj_name fill:transparent,stroke-width:0px
+  style proj1 fill:#1168bd,stroke-width:0px
+  style proj2 fill:#1168bd,stroke-width:0px
+  style proj3 fill:#1168bd,stroke-width:0px
+  
+  style dir_1 fill:#438dd5,stroke-width:0px
+  style dir_2 fill:#438dd5,stroke-width:0px
+  style dir_3 fill:#438dd5,stroke-width:0px
+  style dir_4 fill:#438dd5,stroke-width:0px
+  style dir_5 fill:#438dd5,stroke-width:0px
+
+  style autosync_b fill:#438dd5,stroke-width:0px
+  style back_b fill:#438dd5,stroke-width:0px
+  style front_b fill:#438dd5,stroke-width:0px
+  style transcoder_b fill:#438dd5,stroke-width:0px
+  style scanner_b fill:#438dd5,stroke-width:0px
+
+  style autosync_i1 fill:#85bbf0,stroke-width:0px
+  style back_i1 fill:#85bbf0,stroke-width:0px
+  style back_i2 fill:#85bbf0,stroke-width:0px
+  style front_i1 fill:#85bbf0,stroke-width:0px
+  style transcoder_i1 fill:#85bbf0,stroke-width:0px
+  style scanner_i1 fill:#85bbf0,stroke-width:0px
+  style scanner_i2 fill:#85bbf0,stroke-width:0px
+```
+
+# C4 Diagrams
+Diagrams that focus on capturing project from a high level point of view. Context, Container, Component, Code
+
+## Context
+```mermaid
+C4Context
+  UpdateLayoutConfig($c4ShapeInRow="2", $c4BoundaryInRow="2")  
+  
+  title Context Diagram for Kyoo
+  
+  Person(user, "User")
+  System(kyoo, "Kyoo", "")
+  System_Ext(media, "MediaLibrary", "")
```

Comment:
I think defining those terms would be helpful

Reply:
We can definitely add a glossary.  These are software architecture diagrams, so there is some expectation of knowledge.

Mermaid is still working on adding a legends feature for c4 diagrams.  If those were added I would describe the protocol and ports each relationship made.